### PR TITLE
Use DateTime->set_locale instead of ->set to change the locale

### PR DESCRIPTION
**Pull request by autarch**
Requires DateTime 1.04+

### CODE REVIEW

**Comment by autarch** [comment]
Note that using `->set` this way will start warning in a future version of DateTime

**Comment by theory** [comment]
Thanks Dave!

File: Changes [hunk 1]
```diff
@@ -1,5 +1,9 @@
 Revision history for Perl extension App::Sqitch
 
+     - Use DateTime->set_locale instead of ->set to set the locale. Using
+       ->set may actually change the local time unintentionally. Patch by Dave
+       Rolsky. GitHub PR #303.
```
Comment (ilmari) [inline Changes]: ITYM #304.

Comment (theory) [inline Changes]: Yep, thanks, fixed in 5015a3f.

**Pull request merged**